src/mono/mono/component/debugger-agent.c

```diff
@@ -1398,13 +1398,13 @@ transport_handshake (void)
 	sprintf (handshake_msg, "DWP-Handshake");
 
 	do {
-		res = transport_send (handshake_msg, strlen (handshake_msg));
+		res = transport_send (handshake_msg, (int)strlen (handshake_msg));
```

Comment:
I tried updating the type in prototype and callers, C4267 is then issued for `buffer_init`, `buffer_add_int` and `transport->send`. It is going to be more plumbing for those functions and its callers as well. I don't mind doing it, but do you think we should do it as part of this PR?

Reply:
> It is going to be more plumbing for those functions and their callers as well. I don't mind doing it, but do you think we should do it as part of this PR?

Boo. I appreciate your efforts here. I think we need the mono team to make that decision. I prefer to avoid casts when reasonable changes can make them unnecessary - clearly "reasonable" is subjective in this case and something others will need to clarify.

Reply:
We have to be a little bit careful here.  `transport->send`, in particular is part of an extensibility API that xamarin-macios uses.

I think the cast here is fine.